engine/lib/worker.go

```diff
@@ -415,19 +415,25 @@ func (w *DefaultBaseWorker) OpenStorage(ctx context.Context, resourcePath resour
 }
 
 // Exit implements BaseWorker.Exit
-func (w *DefaultBaseWorker) Exit(ctx context.Context, status libModel.WorkerStatus, err error) error {
+func (w *DefaultBaseWorker) Exit(ctx context.Context, status libModel.WorkerStatus, err error) (errRet error) {
+	// Set the errCenter to prevent user from forgetting to return directly after calling 'Exit'
+	defer func() {
+		w.onError(errRet)
+	}()
+
 	if err != nil {
 		status.Code = libModel.WorkerStatusError
 	}
 
 	w.workerStatus.Code = status.Code
 	w.workerStatus.ErrorMessage = status.ErrorMessage
 	w.workerStatus.ExtBytes = status.ExtBytes
-	if err1 := w.statusSender.UpdateStatus(ctx, w.workerStatus); err1 != nil {
-		return err1
+	if errRet = w.statusSender.UpdateStatus(ctx, w.workerStatus); errRet != nil {
+		return errRet
 	}
 
-	return derror.ErrWorkerFinish.FastGenByArgs()
+	errRet = derror.ErrWorkerFinish.FastGenByArgs()
+	return errRet
```

Comment:
```suggestion
	return derror.ErrWorkerFinish.FastGenByArgs()
```

Reply:
From `return errRet` to `return` to avoid useless copy